examples/test_examples.py

```diff
@@ -50,16 +49,101 @@ def flaky_print(*args):
+	"frontend, f",
+	[
+		pytest.param(
+			frontend,
+            f,
+            marks=pytest.mark.skipif(conditionalExampleTests[f[1]].condition is False,
+                                     reason=conditionalExampleTests[f[1]].reason) if f[1] in conditionalExampleTests.keys() else (),
+		)
+		for frontend, f, in itertools.product(installed, files)
+	],
+    ids = [" {} - {} ".format(f[1], frontend) for frontend, f in itertools.product(installed, files)]
+)
+def testExamples(frontend, f, graphicsSystem=None):
+    # runExampleFile(f[0], f[1], sys.executable, frontend)
+
+    name, file = f
+    global path
+    fn = os.path.join(path,file)
+    os.chdir(path)
+    sys.stdout.write("{} ".format(name))
+    sys.stdout.flush()
+    import1 = "import %s" % frontend if frontend != '' else ''
+    import2 = os.path.splitext(os.path.split(fn)[1])[0]
+    graphicsSystem = '' if graphicsSystem is None else "pg.QtGui.QApplication.setGraphicsSystem('%s')" % graphicsSystem
+    code = """
+try:
+    %s
+    import initExample
+    import pyqtgraph as pg
+    %s
+    import %s
+    import sys
+    print("test complete")
+    sys.stdout.flush()
+    import time
+    while True:  ## run a little event loop
+        pg.QtGui.QApplication.processEvents()
+        time.sleep(0.01)
+except:
+    print("test failed")
+    raise
+
+""" % (import1, graphicsSystem, import2)
+    if sys.platform.startswith('win'):
+        process = subprocess.Popen([sys.executable],
+                                    stdin=subprocess.PIPE,
+                                    stderr=subprocess.PIPE,
+                                    stdout=subprocess.PIPE)
+    else:
+        process = subprocess.Popen(['exec %s -i' % (sys.executable)],
+                                   shell=True,
+                                   stdin=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   stdout=subprocess.PIPE)
+    process.stdin.write(code.encode('UTF-8'))
+    process.stdin.close() ##?
+    output = ''
+    fail = False
+    while True:
+        try:
+            c = process.stdout.read(1).decode()
+        except IOError as err:
+            if err.errno == errno.EINTR:
+                # Interrupted system call; just try again.
+                c = ''
+            else:
+                raise
+        output += c
+
+        if output.endswith('test complete'):
+            break
+        if output.endswith('test failed'):
+            fail = True
+            break
+    time.sleep(1)
+    process.kill()
+    #res = process.communicate()
+    res = (process.stdout.read(), process.stderr.read())
+    if fail or 'exception' in res[1].decode().lower() or 'error' in res[1].decode().lower():
+        print(res[0].decode())
+        print(res[1].decode())
+        pytest.fail("{}\n{}\nFailed {} Example Test Located in {} ".format(res[0].decode(), res[1].decode(), name, file), pytrace=False)
+    assert True
```

Comment:
```suggestion
```
ˋassert Trueˋ has no effect at all

Reply:
I'm a pytest noob, haven't used it much before working on here...does pytest report a test as passing if it gets through a function with no assert failures?

Reply:
Yes, as long as no other error is thrown. Pytest does not even notice passing asserts - `assert expression` is an unlatered Python keyword that just throws an `AssertionError` if `expression` is evaluated to be `False`.

Reply:
be0e95a incorporates this change